### ISSUE TITLE
Support piping through image optimization / compression

### ISSUE THREAD

**Issue by cargocultprogramming** [rbuchberger/jekyll_picture_tag#186]
[Enhancement] It would be nice to have an option to pipe resized images through additional optimizers (e.g. https://github.com/tjko/jpegoptim). This could be in the form of a configurable command line (e.g. `jpegoptim --max=85 <imgname.jpg>`). So original images could retain full quality while giving users the option to add additional compression for generated images.

I guess currently imagemagick defaults are used. Maybe simply allowing options to be passed to imagemagick would be a quick way to implement an easy version of this idea.

Not sure if it makes sense to make this more fine grained e.g. by making it possible to choose different options for different image sizes or media presets.

**Comment by rbuchberger** [comment]
You can currently specify image quality in the preset (per file format, and globally), but you're right that otherwise it's the imagemagick defaults. Passing images through arbitrary shell commands shouldn't be too hard; I'll give it some thought!

**Comment by rbuchberger** [comment]
Sorry for the long delay - after some thought and with the new changes I've decided I'm not going to add this. Instead, in version 2 I've exposed basically all of the write options available to libvips. Any image processing beyond what libvips can do is beyond the scope of this plugin.

Thanks for the feedback, though! There's just too many features to add and I only have so much time to do it. If anyone wants to submit a pull request, I'd happily consider it.
